internal/services/jsonrpc/usersvc/service.go

```diff
@@ -736,3 +849,37 @@ func (svc *Service) ListPendingMigrations(ctx context.Context, req *userjson.Lis
 		Migrations: pendingMigrations,
 	}, nil
 }
+
+func (svc *Service) expireChallenges() {
```

Comment:
Since the ratelimit configuration is fairly high(or maybe the Operator configured high), we might have a big challenge map, maybe worth more consideration?

I'm thinking maybe only loop certain number(use ratelimit configure as a factor to adjust this number) of entries, thus reduce the time the lock is held.

Reply:
Or maybe run checkExpire and delete in different batch, we release the lock between the batches

Reply:
Seems maybe a bit premature IMO.

Reply:
> I'm thinking maybe only loop certain number(use ratelimit configure as a factor to adjust this number) of entries, thus reduce the time the lock is held.

Let's chat about this.  I'm not sure how to do this as envisioned, but my feeling is that a `range` over a map` should just go over the entire map as long as you've already acquired the lock.  This loop should be quite quick.